Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/storage/RiskLevelStorage.kt

```diff
@@ -69,6 +70,14 @@ interface RiskLevelStorage {
      */
     val ptDayRiskStates: Flow<List<PresenceTracingDayRisk>>
 
+    /** PRESENCE TRACING RISK RESULT
+     * The newest 2 results.
+     * Used only for analytics
+     * Can be 0-2 entries.
+     * Newest item first.
+     */
```

Comment:
Please use latestAndLastSuccessfulCombinedEwPtRiskLevelResult instead. It has the last successful pt and ew calculation with day risks which are needed to retrieve some of the data you are using

Reply:
val lastResult = riskLevelStorage
            .latestAndLastSuccessfulCombinedEwPtRiskLevelResult
            .first()
            .lastSuccessfullyCalculated
            .ewRiskLevelResult

Reply:
to access ew and pt RiskLevelResult i would need to make them public

Reply:
did this, if anyone complains we can discuss it